contrib/submit-protocol/src/App/ProtocolForm.tsx

```diff
@@ -318,30 +319,37 @@ export default class ProtocolForm extends React.Component<IProtocolProps, IProto
     );
   }
 
-  private fetchConfig = () => {
+  private fetchConfig = async () => {
     const source = this.props.source;
-    if (source && source.jobName && source.user) {
-      fetch(
-        `${this.props.api}/api/v1/user/${source.user}/jobs/${source.jobName}/config`,
-      ).then((res) => {
-        return res.json();
-      }).then((body) => {
+    const pluginId = this.props.pluginId;
+    if (source && source.jobName && source.user && pluginId) {
+      try {
+        const res = await fetch(
+          `${this.props.api}/api/v1/user/${source.user}/jobs/${source.jobName}/config`,
+        );
+        const body = await res.json();
         const protocol = yaml.parse(body);
+        if (protocol.extras.submitFrom !== pluginId) {
+          throw new Error(`Unknown plugin id ${protocol.extras.submitFrom}`);
+        }
         this.setState(
           { protocol },
           () => this.setJobName(
             null as any,
-            `${source.jobName}_clone_${Math.random().toString(36).slice(2, 10)}`,
+            this.getCloneJobName(source.jobName),
```

Comment:
Why split the state update into two here?

Reply:
to reuse `setJobName`, merged

Reply:
Actually you can call 2 `setState`s in one event loop. React will merge them into one update.